library/tasks/LoomUtils.wdl

```diff
@@ -0,0 +1,156 @@
+  }
+
+  parameter_meta {
+    preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+  }
+
+  command {
+    set -euo pipefail
+
+    python3 /tools/create_loom_ss2.py \
+       --qc_files ${sep=' ' smartseq_qc_files} \
+       --rsem_genes_results  ${rsem_gene_results} \
+       --output_loom_path  "${sample_name}.loom" \
+       --sample_id ${sample_name} \
+  }
+
+  runtime {
+    docker: docker
+    cpu: 4  # note that only 1 thread is supported by pseudobam
+    memory: "18 GiB"
+    disks: "local-disk 100 HDD"
+    preemptible: preemptible
+  }
+
+  output {
+    File loom_output = "${sample_name}.loom"
+  }
+}
+
+
+task OptimusLoomGeneration {
+
+  input {
+    #runtime values
+    String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.2" 
+    # name of the sample
+    String sample_id
+    # gene annotation file in GTF format
+    File annotation_file
+    # the file "merged-cell-metrics.csv.gz" that contains the cellwise metrics
+    File cell_metrics
+    # the file "merged-gene-metrics.csv.gz" that contains the  genwise metrics
+    File gene_metrics
+    # file (.npz)  that contains the count matrix
+    File sparse_count_matrix
+    # file (.npy) that contains the array of cell barcodes
+    File cell_id
+    # file (.npy) that contains the array of gene names
+    File gene_id
+    # emptydrops output metadata
+    File empty_drops_result
+    String counting_mode = "sc_rna"
+
+    Int preemptible = 3
+  }
+  
+  meta {
+    description: "This task will converts some of the outputs of Optimus pipeline into a loom file"
+  }
+
+  parameter_meta {
+    preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+  }
+
+  command {
+    set -euo pipefail
+
+    if [ "${counting_mode}" == "sc_rna" ]; then
+        EXPRESSION_DATA_TYPE_PARAM="exonic" 
+        ADD_EMPTYDROPS_DATA="yes"
+    else
+        EXPRESSION_DATA_TYPE_PARAM="whole_transcript"
+        ADD_EMPTYDROPS_DATA="no" 
+    fi
+
+    python3 /tools/create_loom_optimus.py \
+       --empty_drops_file ${empty_drops_result} \
+       --add_emptydrops_data $ADD_EMPTYDROPS_DATA \
+       --annotation_file ${annotation_file}\
+       --cell_metrics ${cell_metrics}\
+       --gene_metrics ${gene_metrics}\
+       --cell_id ${cell_id}\
+       --gene_id  ${gene_id} \
+       --output_path_for_loom "output.loom" \
+       --sample_id ${sample_id} \
+       --count_matrix ${sparse_count_matrix} \
+       --expression_data_type $EXPRESSION_DATA_TYPE_PARAM
+  }
+
+  runtime {
+    docker: docker
+    cpu: 4  # note that only 1 thread is supported by pseudobam
+    memory: "18 GiB"
+    disks: "local-disk 100 HDD"
+    preemptible: preemptible
+  }
+
+  output {
+    File loom_output = "output.loom"
```

Comment:
Can we name this something more descriptive? Perhaps use the sample id that's input to the pipeline as the basename?

Reply:
Done.